ui/src/worker/jobs/fluxTablesToDygraph.ts

```diff
@@ -0,0 +1,121 @@
+import _ from 'lodash'
+
+import {Message} from 'src/worker/types'
+import {fetchData} from 'src/worker/utils'
+import {FluxTable, DygraphValue} from 'src/types'
+
+export interface FluxTablesToDygraphResult {
+  labels: string[]
+  dygraphsData: DygraphValue[][]
+  nonNumericColumns: string[]
+}
+
+const COLUMN_BLACKLIST = new Set([
+  '_time',
+  'result',
+  'table',
+  '_start',
+  '_stop',
+  '',
+])
+
+const NUMERIC_DATATYPES = ['double', 'long', 'int', 'float']
+
+export const fluxTablesToDygraphWork = (
+  tables: FluxTable[]
+): FluxTablesToDygraphResult => {
+  const allColumnNames = []
+  const nonNumericColumns = []
+
+  const tablesByTime = tables.map(table => {
+    const header = table.data[0]
+    const columnNames: {[k: number]: string} = {}
+
+    for (let i = 0; i < header.length; i++) {
```

Comment:
Could use a forEach here instead with early returns instead of a for with continue.

Reply:
since we are looping over large sets of data, using a for loop is faster than a for each. the reason for not having earlier returns and using continues instead is we want to essentially skip that iteration but still move on to the next.